app/src/common/tx.c

```diff
@@ -20,49 +20,72 @@
 
 #include "apdu_codes.h"
 #include "buffering.h"
+#include "buffering_json.h"
 #include "parser.h"
 #include "zxmacros.h"
 
-#if defined(TARGET_NANOX) || defined(TARGET_NANOS2) || defined(TARGET_STAX) || defined(TARGET_FLEX)
+#define TEMPLATE_JSON_BUFFER_SIZE 1024
+#if !defined(TARGET_NANOS)
 #define RAM_BUFFER_SIZE 8192
-#define FLASH_BUFFER_SIZE 16384
-#elif defined(TARGET_NANOS)
+#define FLASH_BUFFER_SIZE 16384 - TEMPLATE_JSON_BUFFER_SIZE
+#else
 #define RAM_BUFFER_SIZE 256
-#define FLASH_BUFFER_SIZE 8192
+#define FLASH_BUFFER_SIZE 8192 - TEMPLATE_JSON_BUFFER_SIZE
 #endif
 
 // Ram
 uint8_t ram_buffer[RAM_BUFFER_SIZE];
 
+tx_type_t tx_type;
+
 // Flash
 typedef struct {
     uint8_t buffer[FLASH_BUFFER_SIZE];
+    uint8_t templete_json[FLASH_BUFFER_SIZE];
```

Comment:
Is this size correct? Shouldn't we use TEMPLATE_JSON_BUFFER_SIZE?

Reply:
good catch